web/index.jsp

```diff
@@ -29,6 +29,14 @@ include file="projects.jspf"
 {
     PageConfig cfg = PageConfig.get(request);
     cfg.setTitle("Search");
+    String sourceRootPath = cfg.getSourceRootPath();
+    if(sourceRootPath == null || sourceRootPath.isEmpty()){
+    throw new java.io.FileNotFoundException("Configuration File Not Found");
+    }
+    File sourceRootPathFile = RuntimeEnvironment.getInstance().getSourceRootFile();
+    if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
```

Comment:
@tulinkry I intended to do this actually, There are two exceptions, first one is FileNotFoundException, thrown when source path is null or empty, covers the scenario when source root path is not set,
Second exception is IOException, covers the scenario when file is there but cannot read